base/applications/mspaint/selectionmodel.cpp

```diff
@@ -13,11 +14,31 @@
 /* FUNCTIONS ********************************************************/
 
 SelectionModel::SelectionModel()
+    : m_hDC(CreateCompatibleDC(NULL))
+    , m_hBm(NULL)
+    , m_hMask(NULL)
+    , m_ptStack(NULL)
+    , m_iPtSP(0)
 {
-    m_ptStack = NULL;
-    m_iPtSP = 0;
+    SetRectEmpty(&m_rcSrc);
+    SetRectEmpty(&m_rcDest);
+}
 
-    m_hDC = CreateCompatibleDC(NULL);
+SelectionModel::~SelectionModel()
+{
+    DeleteDC(m_hDC);
+    if (m_ptStack)
```

Comment:
Here you can call instead ResetPtStack().

Reply:
OK.